scanner/src/test/kotlin/ScanCodeTest.kt

```diff
@@ -174,47 +186,21 @@ class ScanCodeTest : WordSpec({
         }
     }
 
-    "getClosestCopyrightStatements()" should {
-        "properly return closest jquery copyright statements" {
-            val resultFile = File("src/test/assets/esprima-2.7.3_scancode-2.2.1.json")
-            val result = jsonMapper.readTree(resultFile)
-            val path = "test/3rdparty/jquery-1.9.1.js"
-            val copyrights = scanner.getCopyrightFindings(result).filter { it.location.path == path }
-
-            scanner.getClosestCopyrightStatements(copyrights, 5) shouldBe sortedSetOf(
-                CopyrightFindings(
-                    "Copyright 2005, 2012 jQuery Foundation, Inc.",
-                    sortedSetOf(TextLocation(path, 8, 10))
-                )
+    "getClosestCopyrightStatements" should {
+        "return exactly the statements within the line threshold" {
+            val lineThreshold = 5
+            val licenseStartLine = 10
+            val copyrightFindings = listOf(
+               createCopyrightFinding("statement1", "path", licenseStartLine - lineThreshold - 1),
+               createCopyrightFinding("statement2", "path", licenseStartLine - lineThreshold),
+               createCopyrightFinding("statement3", "path", licenseStartLine + lineThreshold),
+               createCopyrightFinding("statement4", "path", licenseStartLine + lineThreshold + 1)
             )
-            scanner.getClosestCopyrightStatements(copyrights, 3690) shouldBe sortedSetOf(
-                CopyrightFindings(
-                    "Copyright 2012 jQuery Foundation",
-                    sortedSetOf(TextLocation(path, 3688, 3691))
-                )
-            )
-        }
 
-        "properly return closest mootools copyright statements" {
-            val resultFile = File("src/test/assets/esprima-2.7.3_scancode-2.2.1.json")
-            val result = jsonMapper.readTree(resultFile)
-            val path = "test/3rdparty/mootools-1.4.5.js"
-            val copyrights = scanner.getCopyrightFindings(result).filter { it.location.path == path }
+            val result = scanner.getClosestCopyrightStatements(copyrightFindings, licenseStartLine, lineThreshold)
+                .map { it.statement }
 
-            scanner.getClosestCopyrightStatements(copyrights, 28) shouldBe sortedSetOf(
-                CopyrightFindings(
-                    "Copyright (c) 2005-2007 Sam Stephenson",
-                    sortedSetOf(TextLocation(path, 27, 29))
-                ),
-                CopyrightFindings(
-                    "Copyright (c) 2006 Dean Edwards, GNU Lesser General Public",
-                    sortedSetOf(TextLocation(path, 27, 29))
-                ),
-                CopyrightFindings(
-                    "Copyright (c) 2006-2012 Valerio Proietti",
-                    sortedSetOf(TextLocation(path, 23, 23))
-                )
-            )
+            result shouldContainExactlyInAnyOrder listOf ("statement2", "statement3")
```

Comment:
Actually, I find the original code with its concrete mentions of expected copyright statements more readable.

Reply:
Previously the test setup was done differently. It was not possible at all to see from the code whether the assertion do make any sense. Did you include this into the measure of readability when saying the original code was more readable? Are you actually referring only to line 203?

Actually I think there is a trade of involved between the lines of code the test setup needs and exactness of the assertion. I've choosen to trade if of as is because:

* this is in the middle of a refactoring and things will change, this is more lightweight to change
* verifying the complete instance can be done in a single dedicated tests where most other tests
just verify specific aspects of the instance which limits the test redundancy a bit.

Can you please explain as well a bit more detailed your thoughts on this ?

Reply:
I was talking about the overall impression of the whole test at the very first glance. Before, I found it easier to see what this test is about due to its explicit copyright statement strings. Now it's more abstract, and at least I need to stare longer at the code to get what it does, and esp. that it does the same thing as before.

Reply:
As @mnonnenmacher  is ok with this as is, let's consider the discussion on what actual string values to choose for a good test setup move out of scope of this PR.